pkg/controller/machine/controller.go

```diff
@@ -426,9 +426,18 @@ func (r *ReconcileMachine) setPhase(machine *machinev1.Machine, phase string, er
 		machine.Status.LastUpdated = &now
 		if phase == phaseFailed && errorMessage != "" {
 			machine.Status.ErrorMessage = &errorMessage
+			if machine.Annotations == nil {
+				machine.Annotations = map[string]string{}
```

Comment:
you could save the if:
```
annotations := m.GetAnnotations()
annotations["key"] = "value"
m.SetAnnotations(annotations)
```

Reply:
This will cause a panic on `Annotations` being nil, especially in tests. I'm not sure how to avoid nil check here.

Reply:
I'm missing where exactly would this panic?